app/controllers/users_controller.rb

```diff
@@ -64,4 +67,20 @@ def should_display_user?(user)
   def get_user
     @user = User.friendly.find(params[:id])
   end
+
+  def count_timezone_degrees
+    unless params[:timezone_select].blank?
+      if @current_user && @current_user.longitude
+        lng = @current_user.longitude
+        case params[:timezone_select]
+        when 'Close To My Timezone Area'
+          params[:timezone_filter] = [lng - 7.5, lng + 7.5]
+        when 'Wider Timezone Area'
+          params[:timezone_filter] = [lng - 22.5, lng + 22.5]
+        end
+      else
+        redirect_to :back, alert: "Can't determine your location!"
```

Comment:
i wonder if we should disable the geolocation-based filter controls if there is no longitude value for the current user... with a label or tooltip to explain the reason why the controls are disabled... perhaps we tackle this once the feature is further along and we are in a position to consider user experience...